GitUI/CommandsDialogs/BrowseDialog/DashboardControl/DashboardItem.cs

```diff
@@ -164,5 +164,23 @@ private void OnKeyDown(object sender, PreviewKeyDownEventArgs e)
             if (e.KeyCode == Keys.Enter || e.KeyCode == Keys.Space)
                 OnClick(e);
         }
+
+        /// <summary> 
+        /// Clean up any resources being used.
+        /// </summary>
+        /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
+        protected override void Dispose(bool disposing)
+        {
+            CancelBranchNameLoad();
+
+            if (disposing)
+                _branchNameLoader.Dispose();
+
+            if (disposing && (components != null))
```

Comment:
:bulb: This could be combined with the previous check for `disposing` to simplify the code.